credentialdigger/models/snippet_model/snippet_model.py

```diff
@@ -198,6 +214,13 @@ def _remove_initial_junk(self, snippet):
         """ Remove junk from the beginning of a snippet.
         """
         return re.sub(
-            r"^((\s*|\ *)\@\@.*\@\@(\s*|\ *)|(\s*|\ *)\+(\s*|\ *)|(\s*|\ *)\-(\s*|\ *)|(\s*|\ *))",
+            r'^((\s*|\ *)\@\@.*\@\@(\s*|\ *)|(\s*|\ *)\+(\s*|\ *)|(\s*|\ *)\-(\s*|\ *)|(\s*|\ *))',
             "",
             snippet).strip()
+
+    def _check_private_key(self, snippet):
+        """ Check if this snippet is a private key
```

Comment:
> Moreover, you're not checking if the snippet is a private key itself, but only if it's the header of a private key

Yes, we are only checking if this is a header of a private key. Limited by the performance of the Extractor, it never did successfully extract correctly any private key. That said, we now are only testing if we are dealing with a private key's header or not.

On the other hand, I am not in favor of using the Extractor/Classifier for this one task. I believe that it would be better to use a rule such as [`-----BEGIN (\s|.*)PRIVATE KEY-{3,}\n([\s\S]*?)\n-{3,}END (\s|.*)PRIVATE KEY-----`](https://regex101.com/r/zyFPxJ/1) to find out if we are dealing with a private key or not. I made this regex to detect:
- the type of the private key (i.e RSA)
- the contents

We have to find a way to tell that we found a private key, then instead of passing it to the snippet model, we classify it directly as an FP.